lib/file_transfer_agent/encrypt_util.js

```diff
@@ -125,206 +124,278 @@ function EncryptUtil(encrypt, filestream, temp) {
+      dataFile: encryptedFilePath
     };
   };
 
   /**
-  * Decrypt file using AES algorithm.
-  *
-  * @param {Object} encryptionMaterial
-  * @param {String} inFileName
-  * @param {String} tmpDir
-  * @param {Number} chunkSize
-  *
-  * @returns {String}
-  */
-  this.decryptFile = async function (metadata, encryptionMaterial, inFileName,
+   * Decrypt file using AES algorithm.
+   */
+  this.decryptFile = async function (metadata, encryptionMaterial, inputFilePath,
+    tmpDir = null, chunkSize = blockSize * 4 * 1024) {
+    return await this.decryptFileCBC(metadata, encryptionMaterial, inputFilePath, tmpDir, chunkSize);
+  };
+
+  this.decryptFileCBC = async function (metadata, encryptionMaterial, inputFilePath,
     tmpDir = null, chunkSize = blockSize * 4 * 1024) {
-    // Get key and iv from metadata
     const keyBase64 = metadata.key;
     const ivBase64 = metadata.iv;
 
-    // Get decoded key from base64 encoded value
-    const decodedKey = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
-    const keySize = decodedKey.length;
+    const decodedKek = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
 
-    // Get key bytes and iv bytes from base64 encoded value
     const keyBytes = new Buffer.from(keyBase64, BASE64);
     const ivBytes = new Buffer.from(ivBase64, BASE64);
+    const keyDecipher = crypto.createDecipheriv(AES_ECB.cipherName(decodedKek.length), decodedKek, null);
+    const fileKey = performCrypto(keyDecipher, keyBytes);
 
-    // Create temp file
-    let tempOutputFileName;
-    let tempFd;
-    await new Promise((resolve, reject) => {
-      tmp.file({ dir: tmpDir, prefix: path.basename(inFileName) + '#' }, (err, path, fd) => {
+    const dataDecipher = crypto.createDecipheriv(AES_CBC.cipherName(fileKey.length), fileKey, ivBytes);
+    return await performFileStreamCrypto(dataDecipher, tmpDir, inputFilePath, chunkSize);
+  };
+
+  //TODO: Add proper usage when feature is ready (SNOW-940981)
+  this.decryptFileGCM = async function (metadata, encryptionMaterial, inputFilePath, tmpDir = null) {
+    const keyBase64 = metadata.key;
+    const keyIvBase64 = metadata.keyIv;
+    const dataIvBase64 = metadata.iv;
+    const dataAadBase64 = metadata.dataAad;
+    const keyAadBase64 = metadata.keyAad;
+
+    const decodedKek = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
+    const keyBytes = new Buffer.from(keyBase64, BASE64);
+    const keyIvBytes = new Buffer.from(keyIvBase64, BASE64);
+    const dataIvBytes = new Buffer.from(dataIvBase64, BASE64);
+    const dataAadBytes = new Buffer.from(dataAadBase64, BASE64);
+    const keyAadBytes = new Buffer.from(keyAadBase64, BASE64);
+
+    const fileKey = this.decryptGCM(keyBytes, decodedKek, keyIvBytes, keyAadBytes);
+
+    const fileContent = await new Promise((resolve, reject) => {
+      fs.readFile(inputFilePath, (err, data) => {
```

Comment:
why we are not using fsPromises there?

Reply:
I decided to make a new issue for it (SNOW-1814883)